python/update_package_properties.py

```diff
@@ -0,0 +1,301 @@
+            reader.seek(0, 0)
+            old_changelog = reader.read()
+            changelog = f"{debian_latest_changelog}{old_changelog}"
+            reader.seek(0, 0)
+            reader.write(changelog)
+        else:
+            print("Already version in the debian changelog")
+
+
+@validate_parameters
+def update_pkgvars(version: is_version(non_empty(no_whitespaces(non_blank(str)))), fancy: bool,
+                   fancy_release_count: non_negative(int), templates_path: str, pkgvars_path: str) -> None:
+    env = get_template_environment(templates_path)
+
+    version_str = get_version_number(version, fancy, fancy_release_count)
+
+    template = env.get_template('pkgvars.tmpl')
+
+    pkgvars_content = template.render(version=version_str)
+    with open(f'{pkgvars_path}/pkgvars', "w") as writer:
+        writer.write(pkgvars_content)
+
+
+def get_rpm_changelog_history(project_name: str, spec_file_path: str) -> str:
+    with open(spec_file_path, "r") as reader:
+        spec_content = reader.read()
+        changelog_index = spec_content.find("%changelog")
+        changelog_content = spec_content[changelog_index + len("%changelog") + 1:]
+
+    return changelog_content
+
+
+def get_rpm_header(project_name: str, microsoft_email: str, name_surname: str, project_version: str,
+                   is_fancy_versioned: bool,
+                   fancy_version_number: int, changelog_date: date):
+    formatted_date = changelog_date.strftime("%a %b %d %Y")
+    return f"* {formatted_date} - {name_surname} <{microsoft_email}> {get_version_number_with_project_name(project_name, project_version, is_fancy_versioned, fancy_version_number)} "
+
+
+def get_debian_trailer(microsoft_email: str, name_surname: str, changelog_date: date):
+    formatted_date = changelog_date.strftime("%a, %d %b %Y %H:%M:%S %z ")
+    return f" -- {name_surname} <{microsoft_email}>  {formatted_date} \n "
+
+
+def get_rpm_changelog_from_actual_changelog(project_name: str, project_version: str, latest_changelog: str,
+                                            microsoft_email: str,
+                                            name_surname: str, fancy: bool, fancy_version_number: int,
+                                            spec_file_path: str):
+    rpm_changelog_history = get_rpm_changelog_history(project_name, spec_file_path)
+    changelog_histories = rpm_changelog_history.splitlines()
+    rpm_changelog = ""
+    if len(changelog_histories) < 1 or not (project_version in changelog_histories[1]):
+        lines = latest_changelog.splitlines()
+        lines[0] = get_rpm_header(project_name, microsoft_email, name_surname, project_version, fancy,
+                                  fancy_version_number, date.today())
+        rpm_changelog = rpm_changelog.join([str(elem) + "\n" for elem in lines])
+        rpm_changelog = f"{rpm_changelog}{rpm_changelog_history}"
+    else:
+        rpm_changelog = rpm_changelog_history
+
+    return rpm_changelog
```

Comment:
I think this can be simplified quite a bit by doing something like this:

```suggestion
    rpm_changelog_history = get_rpm_changelog_history(project_name, spec_file_path)
    changelog_histories = rpm_changelog_history.splitlines()
    if len(changelog_histories) > 1 and project_version in changelog_histories[1]:
        return rpm_changelog_history
        
    header = get_rpm_header(project_name, microsoft_email, name_surname, project_version, fancy,
                              fancy_version_number, date.today())
    return '\n'.join([header, latest_changelog, rpm_changelog_history]"
```

Reply:
Fixed as you suggested with a small change (
```suggestion
    return '\n'.join([header, latest_changelog, rpm_changelog_history])
```